src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,25 +550,51 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
-                Else
-                    Exit While
+            Dim atNewLine As Boolean
+            Dim FoundCommentAfterUnderscore As Boolean = False
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
+                While CanGet(Here)
+                    ch = Peek(Here)
+                    If IsWhitespace(ch) Then
+                        Here += 1
+                    Else
+                        Exit While
+                    End If
+                End While
+                If CanGet(Here) AndAlso IsSingleQuote(Peek(Here)) Then
+                    If CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                        If Here > 1 Then
+                            tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                        End If
+                        ScanCommentIfAny(tList)
```

Comment:
Parsing the comment after the line continuation should not be conditional on the language version.
The same syntax tree should be produced regardless of language version. The only difference between old vs. new language version should be a diagnostic attached in the tree.

Reply:
I left them separate because I want to figure out how to provide 3 separate errors, missing space before _, Comment in versions before V16 and something is after the _ that is not a comment. Today there is just one error for all issues and it will not help someone discover the new feature. Also I was trying not to change anything in the non-comment path to avoid introducing a bug and every time I try to merge the two paths some test breaks. Its still a work in progress.

Reply:
@paul1956 the concern is about ``` If CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then```

The point is that you should not conditionally process your input.  if you see ```_ ' whatever```, then always lex out comment.  *Then* once you've done that, check if this feature is allowed.  If not, add a diagnostic to the trivia.  Finally, add the comment to the trivia list.

Reply:
I have updated the code to have 2 paths, old which is identical to 15.5 and new which when V16 offers new functionality if there is a valid comment after _ or resorts to old code and reports same errors as V15.5. I have added 2 else statements with comments only. For these two else statement new errors are needed.